src/askvm/lib/resource.ts

```diff
@@ -46,4 +46,26 @@ export function resource<T, A extends any[]>(
   return new Resource(...options);
 }
 
+/**
+ * Resource factory generates the resource wrappers around all the methods which are on `whitelist` of the `module` passed. Implementation of https://github.com/CatchTheTornado/askql/issues/580 for limited JS function calls inside AskScript
+ * @param module JS object which methods will be wrapped as resource handlers
+ * @param whitelist list of strings including function/method names of the module to be wrapped out
+ */
+export function factory(
+  jsModule: any,
+  whitelist: string[]
+): Record<string, Resource<any, any>> {
+  const res: Record<string, Resource<any, any>> = {};
+  Object.keys(jsModule).forEach((k) => {
+    if (whitelist.includes(k)) {
```

Comment:
I am wondering how big the library could be and how suboptimal it would be if there were thousands of keys in `jsModule`.  In the future, we might want to iterate rather through `whitelist` instead, but premature optimization is evil so let's keep it like this for now.

Reply:
Fixed